src/main/java/io/github/bbortt/event/planner/service/ProjectService.java

```diff
@@ -129,6 +133,22 @@ public Project create(CreateProjectDTO createProjectDTO) {
             .orElseThrow(() -> new IllegalArgumentException("Error while persisting project!"));
     }
 
+    @Transactional(readOnly = true)
+    @PreAuthorize("isAuthenticated()")
+    public boolean hasAccessToProject(Long projectId, String... roles) {
+        if (SecurityUtils.isCurrentUserInRole(AuthoritiesConstants.ADMIN)) {
+            return true;
+        }
+
+        return roleRepository.hasAnyRoleInProject(
+            projectRepository
+                .findById(projectId)
+                .orElseThrow(() -> new EntityNotFoundException("Project with id " + projectId + " not found")),
+            SecurityUtils.getCurrentUserLogin().orElseThrow(() -> new IllegalArgumentException("No user Login found!")),
+            Arrays.asList(roles)
+        );
```

Comment:
you can use one single return statement with a logical AND.

Reply:
Ein AND würde einen schönen Bug einführen :P
Und ich finde es mit early return statements okay. Weil wenn man Superadmin ist, ist es klar, dass man Zugriff hat und man muss den ganzen anderen Code nicht lesen.

Reply:
OR : )
naja, beim or ist es auch klar. wenn die erste Bedingung nicht erfüllt ist brichts ja ab. als so:
```java
bedingung 1
  || bedingung 2
```
finde ich recht leserlich. aber war nur ein kommentar, ist ja logisch korrekt.